File: frontend/src/api/classifier.ts
import api from './client'

export interface RoutingResponse {
  recommended_specialty_id: number
  recommended_specialty_name: string
  confidence: number
  reasoning: string
  alternative_specialties: {
    specialty_id: number
    specialty_name: string
    confidence: number
  }[]
}

export interface Specialty {
  id: number
  name: string
}

export async function routeComplaint(complaintText: string): Promise<RoutingResponse> {
  const { data } = await api.post('/api/v1/classifier/route-complaint', { complaint_text: complaintText })
  return data
}

export async function getSpecialties(): Promise<Specialty[]> {
  const { data } = await api.get('/api/v1/classifier/specialties')
  return data
}

File: backend/app/schemas/classifier.py
from pydantic import BaseModel, Field


class ComplaintRequest(BaseModel):
    complaint_text: str = Field(..., min_length=3, max_length=2000, description="Жалоба пациента в свободной форме")


class AlternativeSpecialty(BaseModel):
    specialty_id: int
    specialty_name: str
    confidence: float = Field(ge=0.0, le=1.0)


class RoutingResponse(BaseModel):
    recommended_specialty_id: int
    recommended_specialty_name: str
    confidence: float = Field(ge=0.0, le=1.0, description="Уверенность модели от 0.0 до 1.0")
    reasoning: str = Field(default="", description="Краткое объяснение выбора")
    alternative_specialties: list[AlternativeSpecialty] = Field(default_factory=list)
